MantidQt/MantidWidgets/src/InstrumentView/ComponentActor.cpp

```diff
@@ -62,7 +64,8 @@ namespace MantidQt
 			return  obj &&
 				!getObjCompAssembly() &&
 				!getDetector() &&
-				!boost::dynamic_pointer_cast<const Mantid::Geometry::RectangularDetector>(obj);
+				!boost::dynamic_pointer_cast<const Mantid::Geometry::RectangularDetector>(obj) &&
```

Comment:
The number of dynamic casts required here is not good. That might be an issue for another day.